sdk/storage/ustorageclient/build_all/configs/ustorageclient_build_rules.cmake

```diff
@@ -0,0 +1,144 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#Use solution folders.
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Build with -fPIC always
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
+option(run_valgrind "set run_valgrind to ON if tests are to be run under valgrind/helgrind/drd. Default is OFF" OFF)
+option(compileOption_C "passes a string to the command line of the C compiler" OFF)
+option(compileOption_CXX "passes a string to the command line of the C++ compiler" OFF)
+
+# Relax warnings when applicable
+function(usePermissiveRulesForSamplesAndTests)
+    if(NOT MSVC)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-variable  -Wno-unused-function -Wno-missing-braces -Wno-strict-aliasing")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable  -Wno-unused-function -Wno-missing-braces -Wno-strict-aliasing")
+        if(NOT APPLE AND NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-but-set-variable -Wno-clobbered")
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-but-set-variable -Wno-clobbered")
+        endif()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" PARENT_SCOPE)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}" PARENT_SCOPE)
+    endif()
+endfunction()
+
+# Global variable to know if we are on linux, windows, or macosx.
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    set(ULIB_PAL_DIRECTORY "MSBUILD/X86")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    # Enable valgrind on Linux
+    find_program(MEMORYCHECK_COMMAND valgrind)
+    set(MEMORYCHECK_COMMAND_OPTIONS "--leak-check=full --error-exitcode=1")
+    set(ULIB_PAL_DIRECTORY "GCC/LINUX")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    add_definitions(-DMACOSX)
+    set(ULIB_PAL_DIRECTORY "GCC/IOS")
+endif()
+
+#Enable testing for the target
+enable_testing()
+include(CTest)
```

Comment:
you don't need to include(CTest) unless you want the (much more fancy) cdash stuff